elfi/methods/results.py

```diff
@@ -203,6 +205,59 @@ def sample_means_array(self):
         """
         return np.array(list(self.sample_means.values()))
 
+    def save(self, fname=None, kind='csv'):
+        """Save samples in csv, json or pickle file formats.
+
+        Parameters
+        ----------
+        fname : str, required
+            File name to be saved
+        kind : str, optional
+            File format to be saved. Default format is csv
+
+        """
+        import csv
+        import json
+        import pickle
+
+        # All data in self.__dict__: ['method_name', 'outputs', 'parameter_names', 'meta',
+        #                             'samples', 'discrepancy_name', 'weights']
+        data = ['method_name', 'meta', 'samples']
+        dct = dict()
+
+        for key in self.__dict__.keys():
+            if key in data:
+                dct[key] = self.__dict__[key]
```

Comment:
I think this should be the other way: save everything except those separately mentioned (more general). Keys should be excluded only if really necessary e.g. duplicate or crash.